### ISSUE TITLE
Compile error with ```google/protobuf/port_def.inc``` when using prebuilt protobuf

### ISSUE THREAD

**Issue by owt5008137** [open-telemetry/opentelemetry-cpp#424]
**Describe your environment** 

System: CentOS 7
Compiler: GCC 4.8.5
Branch: master
Installed dependencies:
+ Prebuilt protobuf 3.14.0
+ Only run ```git submodule update --init third_party/opentelemetry-proto``` to initialize submodules.

**Steps to reproduce**

Using cmake to build it. 
```bash
mkdir build_jobs_dir
cd build_jobs_dir
cmake .. -DBUILD_TESTING=OFF -DWITH_OTLP=ON -DWITH_PROTOBUF=ON -DCMAKE_INSTALL_PREFIX=<WHERE TO INSTALL> -DCMAKE_FIND_ROOT_PATH=<PROTOBUF INSTALL PREFIX DIR>
cmake --build . -j
cmake --build . -- install
```

**What is the expected behavior?**
Building success and install all include and libraries into ```<WHERE TO INSTALL>```

**What is the actual behavior?**
It shows ```google/protobuf/port_def.inc``` can not be found.

**Additional context**
After I fix this problem locally, it only install header files but none of libraries is installed.

**Comment by lalitb** [comment]
> It shows google/protobuf/port_def.inc can not be found.

Thanks for reporting. We will look into it.

> After I fix this problem locally, it only install header files but none of libraries is installed.

 The current cmake install rule does api installation ( which is all header ). We are in process of documenting the install instructions which would make the process easy.

**Comment by owt5008137** [comment]
@lalitb But I found the examples are also linked with ```opentelemetry_trace``` ,  ```opentelemetry_common``` and etc. Need these libraries also installed?
BTW: opentelemetry-cpp dn not provide cmake config files for cmake users, and some target names are not same as that in bazel BUILD, can I create a PR and add some support for cmake usage and export these library targets with the same name as bazel target?

**Comment by lalitb** [comment]
> BTW: opentelemetry-cpp dn not provide cmake config files for cmake users, and some target names are not same as that in bazel BUILD, can I create a PR and add some support for cmake usage and export these library targets with the same name as bazel target?

@owt5008137 Thanks. We definitely need more hands here. Please feel free to create a PR. cc @maxgolov.

**Comment by owt5008137** [comment]
> > BTW: opentelemetry-cpp dn not provide cmake config files for cmake users, and some target names are not same as that in bazel BUILD, can I create a PR and add some support for cmake usage and export these library targets with the same name as bazel target?
> 
> @owt5008137 Thanks. We definitely need more hands here. Please feel free to create a PR. cc @maxgolov.
I create two PR. #428 only fix the include problem and #432 add cmake config support for all libraries. The PR export all the target with the same name as it in bazel BUILD. 
I didn't change any c++ codes, but codecov shows #432 will decrease coverage? Maybe it need someone to review it?

**Comment by maxgolov** [comment]
@owt5008137 - ignore the code coverage error. Your PR looks okay. I'll add some feedback later this week.

**Comment by ThomsonTan** [comment]
Wondering whether this build error could reproduce based on the latest master? We added grpc++ dependency to OTLP targets which may include protobuf transitively.

**Comment by owt5008137** [comment]
> Wondering whether this build error could reproduce based on the latest master? We added grpc++ dependency to OTLP targets which may include protobuf transitively.

I updated the codes and found ```protobuf::libprotobuf```,```gRPC::grpc++``` are only linked into example and test targets, not ```opentelemetry_proto``` . This problem may still exists. I have rebased #432 to resolve all the problems above-mentioned.

**Comment by owt5008137** [comment]
Solved by merging #432
